### PR TITLE
feat: implement stream thumbnail

### PR DESCRIPTION
**Pull request by duocphungvanNCC**
> `Description:`

**env development:** window
- implement stream thumbnail
- alow add/change/remove thumbnail image
- check max file upload is 10MB
- confirm upload modal 
- need check confirm modal when remove thumbnail
image
image
image

### CODE REVIEW

File: crates/mezon-store/src/channel.rs [hunk 1]
```diff
@@ -3556,6 +3661,9 @@ impl ChannelList {
                 if changed {
                     cx.notify();
                 }
+                if carries_full_channel_state || !e.channel_avatar.is_empty() {
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: Regression: UpdateChannelPrivate emits channel_type but never channel_avatar (api_channel.go:626-641), so any privacy toggle clears the thumbnail on every client. Treat empty channel_avatar as absent here.

File: crates/mezon-store/src/channel.rs [hunk 2]
```diff
@@ -3134,6 +3135,107 @@ impl ChannelList {
...
+            return Task::ready(Err("Cannot manage stream thumbnail".into()));
+        }
+        let api = self.api.clone();
+        cx.spawn(async move |this, cx| {
+            api.update_channel_desc(
+                clan_id.get(),
+                channel_id.get(),
+                mezon_client::UpdateChannelDescParams {
+                    channel_label: None,
+                    category_id: channel
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: Full-overwrite UpdateChannelDesc from cache that never learns e2ee changes or cleared topics (handler at 3641-3660) reverts other admins' edits. Refetch the detail before writing or patch e2ee/empty topic in the handler.

File: crates/mezon-ui/src/chat/channel_settings/mod.rs [hunk 3]
```diff
@@ -373,6 +379,16 @@ impl ChannelSettingScreen {
                 self.quick_actions_sub = Some(cx.observe(&tab, |_, _, cx| cx.notify()));
                 self.quick_actions_tab = Some(tab);
             }
+            ChannelSettingsTab::StreamThumbnail => {
+                if self.stream_thumbnail_tab.is_none() {
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/mod.rs]: Lint job is red: clippy::collapsible_match. Use a match guard: ChannelSettingsTab::StreamThumbnail if self.stream_thumbnail_tab.is_none() => { ... }

File: crates/mezon-ui/src/chat/channel_settings/stream_thumbnail_tab.rs [hunk 4]
```diff
@@ -0,0 +1,580 @@
...
+        if self.busy {
+            return;
+        }
+        let tab = cx.entity().downgrade();
+        let modal = cx.new(|_| ThumbnailConfirmation {
+            tab,
+            path,
+            settings: self.settings.clone(),
+        });
+        Shell::global(cx).update(cx, |shell, cx| shell.show_modal(modal.into(), cx));
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/stream_thumbnail_tab.rs:234]: Modal never takes focus and has no menu::Cancel/Confirm, so Esc hits ChannelSettingScreen's Cancel (mod.rs:694) and navigates away under the open modal. Add a FocusHandle, focus on open, wire Cancel/Confirm like quick_actions_tab.rs:975.

File: crates/mezon-ui/src/chat/channel_settings/stream_thumbnail_tab.rs [hunk 5]
```diff
@@ -0,0 +1,580 @@
...
+            return;
+        }
+        self.busy = true;
+        cx.notify();
+        cx.spawn(async move |this, cx| {
+            let validated = cx
+                .background_spawn(async move {
+                    validate_clan_image_file(&path, MAX_BYTES)?;
+                    let bytes = std::fs::read(&path).map_err(|e| e.to_string())?;
+                    image::load_from_memory(&bytes)
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/stream_thumbnail_tab.rs]: Full decode with no Limits just to validate: a 20000x12000 PNG allocates ~500 MB on the background thread. Use ImageReader::with_guessed_format()?.into_dimensions() under decoder_limits(), or drop the decode.

File: crates/mezon-store/src/channel.rs [hunk 6]
```diff
@@ -3134,6 +3135,107 @@ impl ChannelList {
...
+        if let Some(channel) = self
+            .user_channels
+            .get_mut(&channel_id)
+            .filter(|channel| channel.clan_id == clan_id)
+        {
+            changed |= channel.avatar_url != avatar;
+            channel.avatar_url = avatar.to_string();
+        }
+        if changed {
+            self.invalidate_channel_index(clan_id);
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: avatar_url affects no channel position or sidebar row; invalidate_channel_index rebuilds the index and evicts sidebar_rows_cache for nothing. Just cx.notify() like set_channel_muted (3357).

File: crates/mezon-ui/src/chat/channel_settings/stream_thumbnail_tab.rs [hunk 7]
```diff
@@ -0,0 +1,580 @@
...
+            let paths = crate::util::file_dialog::resolve(rx, cx).await;
+            let _ = this.update(cx, |this, cx| {
+                this.busy = false;
+                if let Some(path) = paths.and_then(|paths| paths.into_iter().next()) {
+                    this.prepare(path, cx);
+                }
+                cx.notify();
+            });
+        })
+        .detach();
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/stream_thumbnail_tab.rs]: View-scoped task detached (same at 147 and 215); tab is recreated per channel switch so decodes keep running with no receiver. Hold it in a _task: Option<Task<()>> field like edit_group_modal.rs:108.

File: crates/mezon-ui/src/chat/channel_settings/stream_thumbnail_tab.rs [hunk 8]
```diff
@@ -0,0 +1,580 @@
...
+                    Ok::<_, String>(path)
+                })
+                .await;
+            let _ = this.update(cx, |this, cx| {
+                this.busy = false;
+                match validated {
+                    Ok(path) => this.confirm(Some(path), cx),
+                    Err(error) => {
+                        let locale = this.settings.read(cx).language.clone();
+                        let oversized = error.contains("byte limit");
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/stream_thumbnail_tab.rs]: Classifying by message substring: empty file, unreadable path and decode failures all show 'Only image files are allowed'. Return a typed error or check metadata().len() > MAX_BYTES separately.

File: crates/mezon-ui/src/chat/channel_settings/stream_thumbnail_tab.rs [hunk 9]
```diff
@@ -0,0 +1,580 @@
...
+            busy: false,
+            _subs: subs,
+        }
+    }
+
+    fn choose(&mut self, cx: &mut Context<Self>) {
+        if self.busy || !can_manage_channel(self.clan_id, self.channel_id, cx) {
+            return;
+        }
+        self.busy = true;
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/stream_thumbnail_tab.rs]: busy = true while the picker is merely open shows 'Uploading...' with nothing chosen, and removal reuses the same label though streamThumbnail.buttons.removing exists. Track picker vs save separately.

File: crates/mezon-ui/src/chat/channel_settings/stream_thumbnail_tab.rs [hunk 10]
```diff
@@ -0,0 +1,580 @@
...
+                                .font_weight(FontWeight::SEMIBOLD)
+                                .child(self.t("streamThumbnail.upload.title", cx)),
+                        )
+                        .child(
+                            div()
+                                .text_sm()
+                                .child(self.t("streamThumbnail.upload.description", cx)),
+                        )
+                        .child(
+                            Button::new("thumbnail-select")
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/channel_settings/stream_thumbnail_tab.rs:407]: Button has no on_click so it renders non-interactive (no hover/pressed) and relies on the parent drop-zone click, which still fires when disabled. Give it its own on_click calling choose().

File: crates/mezon-ui/src/chat/stream.rs [hunk 11]
```diff
@@ -999,10 +999,10 @@ fn volume_icon(volume: f32, muted: bool) -> IconName {
     }
 }
 
-fn render_stream_thumbnail(channel: &Channel, _cx: &App) -> AnyElement {
+fn render_stream_thumbnail(channel: &Channel, cx: &App) -> AnyElement {
     let raw = channel.avatar_url.trim();
     if !raw.is_empty() && raw != "0" {
-        let raw = SharedString::from(raw.to_string());
+        let raw = SharedString::from(crate::util::imgproxy::stream_cover_url(cx, raw));
```
Comment (huylexuan44) [inline crates/mezon-ui/src/chat/stream.rs:1005]: stream_cover_url forces @webp; imgproxy defaults to max_animation_frames=1, so animated GIFs (advertised as supported) likely flatten to frame 1. Confirm on the deployed imgproxy or keep the raw URL for gif.

File: crates/mezon-store/src/channel.rs [hunk 12]
```diff
@@ -3534,6 +3636,9 @@ impl ChannelList {
                 }
             }
             RealtimeEvent::ChannelUpdated(e) => {
+                if e.is_error {
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: Backend never sets IsError on ChannelUpdatedEvent, and this early return also drops label/topic/private updates. Remove the guard and the ('', 6, true, ...) test row, or scope it to the avatar apply only.

File: crates/mezon-store/src/channel.rs [hunk 13]
```diff
@@ -3134,6 +3135,107 @@ impl ChannelList {
         })
     }
 
+    pub fn upload_stream_thumbnail_image(
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: Line-for-line copy of ClanList::upload_clan_image (clan.rs:1097-1111); every other upload flow calls that helper. Reuse it and delete this.

File: crates/mezon-store/src/channel.rs [hunk 14]
```diff
@@ -7341,6 +7449,71 @@ mod tests {
...
 
+    #[gpui::test]
+    fn stream_thumbnail_remote_changes_and_partial_events(cx: &mut gpui::TestAppContext) {
+        cx.update(|cx| {
+            let channels = init_channel_list(cx);
+            channels.update(cx, |channels, cx| {
+                channels.apply_clan_structure(ClanId(1), structure_with_two_channels(), None, cx);
+                for (avatar, channel_type, is_error, expected) in [
+                    ("thumbnail.png", 6, false, "thumbnail.png"),
+                    ("", 0, false, "thumbnail.png"), // Partial update preserves the image.
```
Comment (huylexuan44) [inline crates/mezon-store/src/channel.rs]: Inline comments here and on 7462 violate the repo's no-comments rule; put the intent in the test name or a labelled tuple.
